### DISCUSSION TITLE
Stripe Payment

### DISCUSSION THREAD

**Discussion by tanujrawat95** [saleor/saleor#12149] [Q&A] [answered]
Hi I am trying to integrate stripe with saleor. I have already enables the plugin and set up a dev account in stripe.
I am trying to understand the flow of payment gateway with saleor. I have completed the steps till completeCheckout mutation where i am able to get client secret and to create payment intent. 
Now next is to pass the use card details, perform payment and get the payment status. As per the documents, the server will get a webhook call from stripe for status update. 
I want to know what will be the webhook endpoint and how will integrate it?

**Comment by korycins** [comment, marked as answer]
Hello @tanujrawat95 ,

To process a payment, please follow the instructions outlined in this guide: [https://stripe.com/docs/payments/accept-a-payment?platform=web&ui=elements#web-collect-payment-details](https://stripe.com/docs/payments/accept-a-payment?platform=web&ui=elements#web-collect-payment-details).

Once the payment is processed, Stripe will send a webhook notification to update the payment on the Saleor side. 
Saleor automatically subscribes to Stripe's webhook when you activate the Stripe plugin. For more information on activating Stripe webhooks, please refer to this guide: [https://docs.saleor.io/docs/3.x/developer/available-plugins/stripe#activating-stripe-webhooks](https://docs.saleor.io/docs/3.x/developer/available-plugins/stripe#activating-stripe-webhooks).

If you wish to test the payment processing locally, you will need to use a tool like ngrok to forward the traffic.

**Reply by tanujrawat95** [reply]
Hi @korycins ! Thanks for the reply.
Sorry I am new to payment integration. I wanted to know how the stripe will know where to send the webhook events. Is there any particular endpoint where Stripe will send the notification?
Also I am trying to test it locally but unable to. Is there any documentation available to show how to process the payments locally in Saleor?

**Reply by korycins** [reply]
To use webhooks on your local machine, you need to use a tool such as ngrok to forward traffic, as described in [https://docs.saleor.io/docs/3.x/developer/available-plugins/stripe#activating-stripe-webhooks](https://docs.saleor.io/docs/3.x/developer/available-plugins/stripe#activating-stripe-webhooks).

Even if you have not enabled webhooks, you should still be able to finalize payments with Stripe on your local machine. After Stripe processes the payment, calling checkoutComplete will return a newly created order. Saleor fetches the payment status from Stripe.

You can find declared webhooks for Stripe in the Stripe admin panel. Stripe has a separate section for webhooks and their logs.

Refer to [https://docs.saleor.io/docs/3.x/developer/available-plugins/stripe](https://docs.saleor.io/docs/3.x/developer/available-plugins/stripe) for documentation on how to use the Stripe plugin.

Stripe sends a webhook when a payment-related event occurs. You can find the list of events that Saleor requests when creating the webhook in [https://docs.saleor.io/docs/3.x/developer/available-plugins/stripe#stripe-webhooks](https://docs.saleor.io/docs/3.x/developer/available-plugins/stripe#stripe-webhooks).

**Reply by tanujrawat95** [reply]
Thanks for the help @korycins . I am able to use ngork and hopefully will be able to run stripe successfully.
One thing that I still have in my mind is that when I enable stripe plugin in my local machine, the webhook enpoint field is blank
![image](https://user-images.githubusercontent.com/66406059/222094483-d56c2820-2ad8-47e0-9e62-e335c25de12d.png)

But when i enable the same plugin in saleor cloud, it is getting generated automatically.

Can you please tell why is this happening and Is is required to have webhook endpoint in Stripe Plugin configuration?

**Reply by korycins** [reply]
You need to update your Shop settings to store an ngrok domain that is accessible from the network. By default, it is set to localhost. To change the domain, call `shopDomainUpdate` (see [https://docs.saleor.io/docs/3.x/api-reference/mutations/shop-domain-update](https://docs.saleor.io/docs/3.x/api-reference/mutations/shop-domain-update)). The Stripe plugin will use the provided domain to generate the webhook URL that will be sent to Stripe. The `webhook_endpoint` will be fulfilled with the ID of the webhook, which can be double-checked on the Stripe side.

Note that you do not need to perform this step on Saleor cloud, as the domain is already defined there.

**Reply by tanujrawat95** [reply]
Thanks a lot @korycins . It worked. Wasted a lot of time over this. 
In documents, it is asking to update site setting configuration but didnt mention the steps. Would be great if it were there so anyone can understand it better.
Thanks a lot again

**Reply by korycins** [reply]
Happy to help :) 
I've created the GH issue for the missing part of the docs: https://github.com/saleor/saleor-docs/issues/673

**Reply by tanujrawat95** [reply]
Great! Thanks for the quick updates. Really appreciate
